cli/2024-10-cli-udt.md

```diff
@@ -0,0 +1,232 @@
+- Other interfaces and experiences to interact with user defined resource type definitions and resources are out of scope for this document
+
+## Key scenarios
+<!-- List ~3-7 high level scenarios to clarify the value and point to how we will decompose this big area into component capabilities. We may ultimately have more than one level of scenario. -->
+The key scenarios for UDTs are outlined [here](/architecture/2024-06-resource-extensibility-feature-spec.md). We will decompose these scenarios specific to the CLI experience.
+
+### Scenario 1: Deb defines and deploys the internal messaging service using the CLI
+<!-- One or two sentence summary -->
+Deb, a platform engineer who wants to define and deploy their internal messaging service Plaid as a resource type in Radius. 
+
+Before we start designing the experience, we need to understand the high level terminologies/concept that a user would interact with in the CLI experience.
+
+| Terminology | Description | Example | 
+|-------------|-------------|---------|
+|Resource provider | Refers to a group of resource types under a common namespace. Resource providers are also entities that implement the group of resource types | Mycompany.Messaging or Applications.Core |
+|Resource type | A service or technology that can be modeled as a resource | PostgreSQL or AWS S3 or Internal messaging service |
+|Resource | An instance of a resource type | mydb or mybucket |
+
+#### User Experience
+Below is an illustrative example of how Deb would define the internal messaging service Plaid as a resource type in Radius. 
+
+1. Deb creates the resource provider and the resource type with Radius
+
+    ```bash
+    rad resource-provider create -f Mycompany.Messaging.yaml
+    ```
+    `Mycompany.Messaging.yaml` is the schema definition file that contains the resource type definition for the messaging service Plaid. See the feature spec that covers the [authoring experience for the schema definition](https://github.com/radius-project/design-notes/blob/67dd03056d22bcae2ee9743ffb308c8ac7f012bd/architecture/2024-10-user-defined-types-authoring-experience-feature-spec.md)
+
+2. Deb creates the resource instance for the messaging service Plaid
+
+    ```bash
+    rad resource create Mycompany.Messaging/Plaid myplaid -f myplaid.json
+    ```
+    The resource `myplaid` of type `Mycompany.Messaging/plaidResource`is created and available for use.
+
+4. Deb lists the resource providers and resource types available in Radius
+
+    ```bash
+    rad resource-provider list
+    ```
+    The list of resource providers are displayed.
+
+    ```bash
+    rad resource-type list --resource-provider Mycompany.Messaging
+    ```
+    The list of resource types under the resource provider `Mycompany.Messaging` are displayed.
+
+5. Deb makes changes to the resource type definition of `plaidResource` and adds to the resource provider
+
+    ```bash
+    rad resource-provider create -f Mycompany.Messaging.yaml
```

Comment:
three questions: 
1. Does Deb need to specify the name of the resource type (i.e. `plaidResource`) as a part of this command?
2. Are there any thoughts about overloading `create` to perform both creation and update operations vs. adding another command for `update`?
3. Should this be a `rad resource-type create` command instead?

Reply:
1. Yes updated this to use `rad resource-type` as its more intuitive and we expect users to initially interact via that commans
2. Create is both create and update. This is consistent with other commands in Radius. We do not have separate commands for update.
3. Yes